### PR TITLE
coprocessor: collect output counts for each executor

### PR DESCRIPTION
**Pull request by lamxTyler**
We collect it so that we could this information to adjust statistics.
PTAL @coocood @hanfei1991 @winoros @AndreMouche

### CODE REVIEW

File: src/coprocessor/dag/dag.rs [hunk 1]
```diff
@@ -91,6 +93,9 @@ impl DAGContext {
                     let mut resp = Response::new();
                     let mut sel_resp = SelectResponse::new();
                     sel_resp.set_chunks(RepeatedField::from_vec(chunks));
+                    let mut counts = Vec::with_capacity(self.num_execs);
```
Comment (coocood) [inline src/coprocessor/dag/dag.rs]: I think we can simply use a constant number like 4 for capacity.

**Comment by coocood** [comment]
LGTM

File: src/coprocessor/dag/executor/index_scan.rs [hunk 2]
```diff
@@ -170,6 +174,10 @@ impl Executor for IndexScanExecutor {
         }
     }
 
+    fn collect_output_counts(&mut self, counts: &mut Vec<i64>) {
+        counts.push(self.count - 1);
```
Comment (AndreMouche) [inline src/coprocessor/dag/executor/index_scan.rs]: You'd better set self.count to zero after collect

File: src/coprocessor/dag/executor/index_scan.rs [hunk 3]
```diff
@@ -145,6 +148,7 @@ impl IndexScanExecutor {
 
 impl Executor for IndexScanExecutor {
     fn next(&mut self) -> Result<Option<Row>> {
+        self.count += 1;
```
Comment (AndreMouche) [inline src/coprocessor/dag/executor/index_scan.rs]: You'd better increase count only when meets some value since we will support stream in the future.

**Comment by hicqu** [comment]
LGTM.

**Pull request merged**